mountpoint-s3/tests/common/fuse.rs

```diff
@@ -75,6 +75,20 @@ impl TestSessionConfig {
     }
 }
 
+// Holds resources for the testing session and cleans them on drop.
+pub struct TestSession {
+    pub mount_dir: TempDir,
+    pub session: BackgroundSession,
+    pub test_client: TestClientBox,
+}
```

Comment:
Not blocking: we could make these private if we really wanted to, with `mount_dir(&self) -> &Path` and `test_client(&mut self) -> &mut TestClientBox`.

I don't believe the session is even used by anyone (but needs to stick around).

Reply:
It's a good point but since we try to borrow `TestSession` as both immutable (via `mount_dir(&self)`) and mutable (via `test_client(&mut self)`) it causes `cannot borrow test_session as mutable because it is also borrowed as immutable`. I think it requires more refactoring to properly seperate them.

Reply:
Ah, makes sense. I was thinking we probably want to add interior mutability to the mock client but all good for now.